Switch I-O Data and Apresia devices to new RTL8231 driver

svanheule opened the pull request:
Convert two devices to the new RTL8231 driver, originally submitted by @musashino205 

The Apresia switch can make use of the MDIO infrastructure for the RTL8231's reset line. A HW debounce config is also added for the the reset key.

The I-O DATA switch is a simple conversion and should not result in behavioural changes.

svanheule commented:
I went ahead and merged the I-O Data device with some other (simple) conversions.

@musashino205 If you have time, could you have a look at the remaining commit for the Apresia device? The button debounce isn't really required, I guess. If the user wants to restart the device, it's not too much of a problem if the signal is sent more than once.

musashino205 commented:
Thank you for the work, LGTM.

> @musashino205 If you have time, could you have a look at the remaining commit for the Apresia device? The button debounce isn't really required, I guess.

I haven't had time to test it, but I think that's probably true.

svanheule committed:
realtek: debounce reset key of APLGS120GT-SS

Since the GPIO used for the reset key is capable of hardware debouncing,
add the (only supported) debounce interval of 100ms.

Signed-off-by: Sander Vanheule <sander@svanheule.net>

svanheule commented:
I've merged the RTL8231 migration, but left out the reset key debounce in a separate commit. I'll close the PR for now, but we  can revisit later on if needed.
